### PR TITLE
chore(web): unit test polishing pass

### PR DESCRIPTION
**Pull request by jahorton**
This should help make Web's unit testing a little 'snappier', especially when things fail.

Changes made:

1. Keyboard sequence reproduction has been retooled to operate based on `Promise`s, allowing unhandled `Promise`-internal errors to be caught by the test framework.
    - I noticed that when a test dependent on reproducing a keystroke sequence failed, sometimes the test's full timeout would be enforced- likely because we weren't triggering the test-function's callback.  In particular, if the error was raised within a `Promise`'s operation, and not within the synchronous test-setup process.
    - I have seen a few occasions where KMW builds have timed out; this may help mitigate such cases.  No guarantees though, as those builds didn't actually report _any_ failed tests.
        - Like this one:  https://build.palaso.org/viewLog.html?buildId=337434&buildTypeId=Keymanweb_TestPullRequests&tab=buildResultsDiv
    - That said, with the length of wait I had locally before the test fully timed out, and the magnified timeout timer we've got set up for BrowserStack remote testing, I wouldn't be surprised if the cause is related.

2. Now that such tests can fully utilize `Promise`s, I reordered related tests to be more "linear" in their specifications, thus easier to maintain.
3. A similar issue existed for some of the event-oriented tests, so I retooled them to be fully synchronous.
    - They'd complete quickly when successful, but the way they were written, the full timeout would be required should the test fail.  And, upon further inspection, those particular tests should've been synchronous to begin with.
    - Example build: https://build.palaso.org/viewLog.html?buildId=340612&buildTypeId=Keymanweb_TestPullRequests
4. Removed any conditioning upon the existence of `InputEvent` - that was an IE-testing holdover.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/unit_tests/cases/events.js [hunk 1]
```diff
@@ -48,9 +47,11 @@ describe('Event Management', function() {
     if(focusEvent) {
       ele.dispatchEvent(focusEvent);
     }
+
+    assert.isNull(ele.onchange);
```
Comment (mcdurdin) [inline web/unit_tests/cases/events.js:51]: I see what this is doing but it's a bit indirect. Perhaps a comment or assertion message would help.

File: web/unit_tests/cases/events.js [hunk 2]
```diff
@@ -48,9 +47,11 @@ describe('Event Management', function() {
     if(focusEvent) {
```
Comment (mcdurdin) [inline web/unit_tests/cases/events.js:47]: Can `focusEvent` ever be falsy here? I'd not think so -- in which case the `if()` is unnecessary and unhelpful

File: web/unit_tests/cases/events.js [hunk 3]
```diff
@@ -83,18 +83,14 @@ describe('Event Management', function() {
       focusEvent.initFocusEvent("blur", true, false, ele.ownerDocument.defaultView, 0, ele);
```
Comment (mcdurdin) [inline web/unit_tests/cases/events.js:83]: This branch is IE-only. We should destroy it.

File: web/unit_tests/cases/events.js [hunk 4]
```diff
@@ -83,18 +83,14 @@ describe('Event Management', function() {
...
     }
 
-    if(focusEvent)
+    if(focusEvent) {
```
Comment (mcdurdin) [inline web/unit_tests/cases/events.js:86]: Again, can't see this can ever be falsy

**Review by mcdurdin — APPROVED** [review]
LGTM. Suggest removing a few more IE-isms. There may be others hidden away similarly.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.77-alpha](https://keyman.com/downloads/releases/alpha/16.0.77)

**Pull request merged**